Dynatrace/dynatrace-operator
# Add operator deployment test in cloudNativeFullStack mode with running istio

# Description

New CloudNative istio test is based on existing CloudNative test. 

The following configuration changes are applied to enable Istio injection 
- _istio-injection_ label is added to the application namespace
- _istio-injection_ label is added to the _dynatrace_ namespace
- _enableIstio_ property is set to true

The following additional checks are performed
- Istio service mesh is up and running
- _istio-init_ container is injected into application pods
- _istio-init_ container is injected into operator pods
- _istio virtualservice_ related to ApiUrl is created
- _istio serviceentry_ related to ApiUrl is created

Additionally
- test/e2e/cloudnative/istio target has been added

## How can this be tested?
1) Setup k8s cluster with Istio service mesh enabled.
2) Create _testdata/secrets/multi-tenant.yaml_ (see test/testdata/secrets-samples).
3) Create _testdata/secrets/single-tenant.yaml_(see test/testdata/secrets-samples).
4) Use make command to start a test.
```
make test/e2e/cloudnative/istio 
make test/e2e/cloudnative
...
```

## Checklist
